public/app/features/alerting/unified/components/rules/central-state-history/HistoryEventsList.test.tsx

```diff
@@ -27,4 +50,92 @@ describe('HistoryEventsList', () => {
       'June 14 at 06:38:30alert1alertnamealert1grafana_folderFOLDER Ahandler/alerting/*'
     );
   });
+
+  it('should render the list correctly filtered by from and to state ', async () => {
+    render(
+      <HistoryEventsList
+        valueInLabelFilter={''}
+        valueInStateFromFilter={StateToFilterValues.firing}
+        valueInStateToFilter={StateToFilterValues.normal}
+        addFilter={jest.fn()}
+      />
+    );
+    await waitFor(() => {
+      expect(ui.loadingBar.query()).not.toBeInTheDocument();
+    });
+    expect(ui.rowHeader.getAll()).toHaveLength(1);
+    expect(ui.rowHeader.getAll()[0]).toHaveTextContent(
+      'June 14 at 06:38:30alert2alertnamealert2grafana_folderFOLDER Ahandler/alerting/*'
+    );
+  });
+
+  it('should render the list correctly filtered by state to', async () => {
+    render(
+      <HistoryEventsList
+        valueInLabelFilter={''}
+        valueInStateToFilter={StateToFilterValues.firing}
+        valueInStateFromFilter={StateFromFilterValues.all}
+        addFilter={jest.fn()}
+      />
+    );
+    await waitFor(() => {
+      expect(ui.loadingBar.query()).not.toBeInTheDocument();
+    });
+    expect(ui.rowHeader.getAll()).toHaveLength(2);
+    expect(ui.rowHeader.getAll()[0]).toHaveTextContent(
+      'June 14 at 06:39:00alert2alertnamealert2grafana_folderFOLDER Ahandler/alerting/*'
+    );
+    expect(ui.rowHeader.getAll()[1]).toHaveTextContent(
+      'June 14 at 06:38:30alert1alertnamealert1grafana_folderFOLDER Ahandler/alerting/*'
+    );
+  });
+  it('should render the list correctly filtered by state from', async () => {
+    render(
+      <HistoryEventsList
+        valueInLabelFilter={''}
+        valueInStateFromFilter={StateToFilterValues.firing}
+        valueInStateToFilter={StateToFilterValues.all}
+        addFilter={jest.fn()}
+      />
+    );
+    await waitFor(() => {
+      expect(ui.loadingBar.query()).not.toBeInTheDocument();
+    });
+    expect(ui.rowHeader.getAll()).toHaveLength(1);
+    expect(ui.rowHeader.getAll()[0]).toHaveTextContent(
+      'June 14 at 06:38:30alert2alertnamealert2grafana_folderFOLDER Ahandler/alerting/*'
+    );
+  });
+
+  it('should render the list correctly filtered by label and state to', async () => {
+    render(
+      <HistoryEventsList
+        valueInLabelFilter={'alertname=alert1'}
+        valueInStateToFilter={StateToFilterValues.firing}
+        valueInStateFromFilter={StateFromFilterValues.all}
+        addFilter={jest.fn()}
+      />
+    );
+    await waitFor(() => {
+      expect(ui.loadingBar.query()).not.toBeInTheDocument();
+    });
+    expect(ui.rowHeader.getAll()).toHaveLength(1);
+    expect(ui.rowHeader.getAll()[0]).toHaveTextContent(
+      'June 14 at 06:38:30alert1alertnamealert1grafana_folderFOLDER Ahandler/alerting/*'
+    );
+  });
+
+  it('should handle an empty list when no events match the filter criteria', async () => {
```

Comment:
(I think we need an empty state as well, when the table has been filtered to no results? Feels like this test would be asserting on something other than the loading bar disappearing)

Reply:
I forgot adding the expect for the list not being rendered. Adding it!